### PR TITLE
algorithms/GenericFunctions: `x == 0` -> `iszero(x)`

### PR DESCRIPTION
**Pull request by nsajko**
Updates #1868

### CODE REVIEW

File: src/algorithms/GenericFunctions.jl [hunk 1]
```diff
@@ -118,7 +118,7 @@ function powermod(a::T, n::Integer, m::T) where T <: RingElem
       return internal_powermod(a, n, m)
    elseif n == 1
```
Comment (fingolfin) [inline src/algorithms/GenericFunctions.jl:119]: You might as well also do 
```suggestion
   elseif isone(n)
```

File: src/algorithms/GenericFunctions.jl [hunk 2]
```diff
@@ -25,7 +25,7 @@ function ^(a::T, n::Integer) where T <: RingElem
       return internal_power(a, n)
    elseif n == 1
       return deepcopy(a)
-   elseif n == 0
```
Comment (thofma) [inline src/algorithms/GenericFunctions.jl:28]: no, I don't think this is better

File: src/algorithms/GenericFunctions.jl [hunk 3]
```diff
@@ -118,7 +118,7 @@ function powermod(a::T, n::Integer, m::T) where T <: RingElem
       return internal_powermod(a, n, m)
    elseif n == 1
       return mod(a, m)
-   elseif n == 0
```
Comment (thofma) [inline src/algorithms/GenericFunctions.jl:121]: no, as above

**Comment by thofma** [comment]
Sorry, I should have just written it here once. I don't think this is an improvement.

**Comment by thofma** [comment]
Ah sorry, I was not thinking about BigInt case. Everything is fine then.

**Review by lgoettgens — APPROVED** [review]

**Pull request merged**